sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -0,0 +1,201 @@
+  getStudentToken,
+  mockUserLoginInfo,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeStudentAppeal,
+  createFakeStudentAppealRequest,
+  getProviderInstanceForModule,
+  saveFakeApplicationDisbursements,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { TestingModule } from "@nestjs/testing";
+import { ApplicationStatus, StudentAppealStatus } from "@sims/sims-db";
+import { StudentAppealAPIInDTO } from "../../models/student-appeal.dto";
+import { AppStudentsModule } from "../../../../app.students.module";
+import { FormService } from "../../../../services";
+
+describe("StudentAppealStudentsController(e2e)-submitStudentAppeal", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let appModule: TestingModule;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it(
+    "Should create a student appeal for financial information change when student submit " +
+      "an appeal and does not have any other pending appeal.",
+    async () => {
+      // Arrange
+      // Create student to submit application.
+      const student = await saveFakeStudent(appDataSource);
+      // Create application to request change.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+        },
+        { applicationStatus: ApplicationStatus.Completed },
+      );
+      // Prepare the data to request a change of financial information.
+      const financialInformationFormName = "studentfinancialinformationappeal";
+      const financialInformationData = {
+        programYear: application.programYear.programYear,
+        studentNewTaxReturnIncome: 8000,
+        studentNewHaveDaycareCosts12YearsOrOver: "no",
+        studentNewHaveDaycareCosts11YearsOrUnder: "no",
+      };
+      const payload: StudentAppealAPIInDTO = {
+        studentAppealRequests: [
+          {
+            formName: financialInformationFormName,
+            formData: financialInformationData,
+          },
+        ],
+      };
+      // Mock user service to return the saved student.
+      await mockUserLoginInfo(appModule, student);
+      // Get any student user token.
+      const studentToken = await getStudentToken(
+        FakeStudentUsersTypes.FakeStudentUserType1,
+      );
+      // Mock the form service to validate the dry-run submission result.
+      // TODO: Form service must be hosted for E2E tests to validate dry run submission
+      // and this mock must be removed.
+      const formService = await getProviderInstanceForModule(
+        appModule,
+        AppStudentsModule,
+        FormService,
+      );
+      const mockDryRunSubmission = jest.fn().mockResolvedValue({
```

Comment:
As a suggestion I would prefer have the method mocked as below
```
formService.dryRunSubmission = jest.fn().mockResolvedValue({
    valid: true,
    formName: financialInformationFormName,
    data: { data: financialInformationData },
  });
```
and then the expect would read
```
expect(formService.dryRunSubmission).toHaveBeenCalledWith
```
If the above suggestion is not accepted, in a non-related suggestion, I would prefer the variable name to be `dryRunSubmissionMock`.

If both suggestion are not accepted, this would not be a blocker.